src/main/java/com/aws/greengrass/logging/impl/LogManager.java

```diff
@@ -118,33 +120,73 @@ public static void setRoot(Path newPath) {
     }
 
     /**
-     * Sets effective configuration.
-     *
-     * @param loggerConfiguration configuration set in yaml.
+     * If a field is null in the given loggerConfiguration, set it using the value from root logging config.
+     * Effectively inheriting the root config (which is for greengrass.log)
      */
-    public static void setEffectiveConfig(LoggerConfiguration loggerConfiguration) {
-        LogConfig rootConfig = LogConfig.getInstance();
+    public static void setNullFieldsFromRootConfig(LoggerConfiguration loggerConfiguration) {
         if (loggerConfiguration.getFileName() == null || loggerConfiguration.getFileName().trim().isEmpty()) {
-            loggerConfiguration.setFileName(rootConfig.getFileName());
```

Comment:
This was the cause of the bug. File name will not be given in the `loggerConfiguration` so it'll be set to the same as root config which is `greengrass.log`. That's why the config won't apply to component loggers.

Reply:
Thanks for pointing at the exact place of the cause of the bug, Tilo. Super helpful!